src/oscar/apps/dashboard/reports/reports.py

```diff
@@ -19,13 +19,18 @@ class ReportGenerator(object):
     code = ''
     description = '<insert report description>'
     date_range_field_name = None
+    model_class = None
+    queryset = None
 
     def __init__(self, **kwargs):
         self.start_date = kwargs.get('start_date')
         self.end_date = kwargs.get('end_date')
 
-        formatter_name = '%s_formatter' % kwargs['formatter']
+        formatter_name = '%s_formatter' % kwargs.get('formatter', 'HTML')
         self.formatter = self.formatters[formatter_name]()
+        if self.queryset is None:
```

Comment:
I think this check is better done inside the `get_queryset` method itself, so that projects can override behaviour more easily. See [here](https://github.com/django/django/blob/master/django/views/generic/detail.py#L58-L76) for the pattern Django uses.

Reply:
If we do this change, we'd need to add `queryset = self.get_queryset()` in all `generate` method calls, instead of just refer to `self.queryset`, right?

Reply:
What I had in mind was:

`self.queryset = self.get_queryset()` here, and in `get_queryset()` we do:

```
if self.queryset is not None:
    return self.queryset
```

So that we ensure that `get_queryset()` is *always* called, and if a project wants to override (e.g., to add some filtering or select_relateds) then it can do so without worrying about whether `queryset` has been set on the class. What do you think?

Reply:
Yeah, agreed.

Reply:
Ok done.